x-pack/plugins/lens/public/editor_frame_service/editor_frame/state_management.ts

```diff
@@ -282,7 +282,10 @@ export const reducer = (state: EditorFrameState, action: Action): EditorFrameSta
         ...state,
         visualization: {
           ...state.visualization,
-          state: action.newState,
+          state:
+            typeof action.updater === 'function'
+              ? action.updater(state.visualization.state)
+              : action.updater,
```

Comment:
This is a good change, but does it need to be in this PR specifically?

Reply:
This PR was the place we realised about this possible change. Making it here helped fixing a `useEffect` side effect which was affecting the datatable visualization.